src/gui/properties/propertieswidget.cpp

```diff
@@ -311,8 +311,11 @@ void PropertiesWidget::loadTorrentInfos(BitTorrent::TorrentHandle *const torrent
       label_created_by_val->setText(m_torrent->creator());
 
       // List files in torrent
-      PropListModel->model()->setupModelData(m_torrent->info());
-      filesList->setExpanded(PropListModel->index(0, 0), true);
+      BitTorrent::TorrentInfo info = m_torrent->info();
+      libtorrent::file_storage files = info.files();
+      PropListModel->model()->setupModelData(info);
+      if (!(info.filesCount() > 1 && files.name().empty()))
```

Comment:
I assume you want a new wrapped type? Wouldn't that need to also extend TorrentInfo to return that type?
Or extend TorrentInfo with a new function that will do the check internally? eg
```c++
bool TorrentInfo::isSubfolderCreated() const
{
    libtorrent::file_storage files = info.files();
    return !(filesCount() > 1 && files.name().empty())
}
```